0016-consensus-parameters-change-proposal.md

```diff
@@ -0,0 +1,124 @@
+# ADR 0016: Consensus Parameters Change Proposal
+
+## Component
+
+Oasis Core
+
+## Changelog
+
+- 2022-09-15: Initial version
+
+## Status
+
+Proposed
+
+## Context
+
+Currently consensus parameters can only be changed with an upgrade governance
+proposal which is effective but not very efficient. Upgrades require downtime
+during which binaries need to be updated, nodes restarted and synced, consensus
+network version has to be increased etc. We would like to avoid this cumbersome
+procedure and change the parameters of an application as fast and as simple
+as possible without affecting the performance of the consensus layer.
+
+## Decision
+
+Implement governance proposal which changes consensus parameters only.
+
+## Implementation
+
+A new type of governance proposal named `ChangeParametersProposal` should be
+added to the consensus layer.
+
+```golang
+type ProposalContent struct {
+  ...
+  ChangeParameters *ChangeParametersProposal `json:"change_parameters,omitempty"`
+}
+```
+
+The proposal should contain a CBOR-encoded `Document` describing:
+
+- the name of the consensus application the changes apply to and
+
+- parameter changes which should be applied to the application's
+`ConsensusParameters` structure,
+
+and na `Epoch` at which the changes should be applied.
+
+```golang
+// ChangeParametersProposal is a consensus parameters change proposal.
+type ChangeParametersProposal struct {
+  // Document is a consensus parameters change document.
+  Document cbor.RawMessage `json:"document"`
```

Comment:
I believe this should just be:
```suggestion
  Document Document `json:"document"`
```
As there is no reason to have two levels of these.